core/eds.go

```diff
@@ -25,7 +25,7 @@ import (
 // ExtendedDataSquare (EDS). If there are no transactions in the block,
 // nil is returned in place of the eds.
 func extendBlock(data types.Data, appVersion uint64, options ...nmt.Option) (*rsmt2d.ExtendedDataSquare, error) {
-	if app.IsEmptyBlock(data, appVersion) {
+	if app.IsEmptyBlockRef(&data, appVersion) {
```

Comment:
this is a new method that apparently does the same thing with improved performance. See https://github.com/celestiaorg/celestia-app/pull/4063